### ISSUE TITLE
Navigator series visiblity side effect on legend click / show() method

### ISSUE THREAD

**Issue by bhaskar20** [highcharts/highcharts#7129]
#### Expected behavior
a navigator series should stay hidden if explicitly passing visibility option for series
```
navigator: {
  series: {
    data: MSFT,
    visible: false
  },
  adaptToUpdatedData: false
}
```

#### Actual behaviour
On initial rendering, Navigator series in not visible, but when show method is called on the first series of chart, Navigator series is visible without honoring the navigator options.

#### Live demo with steps to reproduce
http://jsfiddle.net/toufu__/4qv278zf/

#### Affected browser(s)
Tested in chrome Version 60.0.3112.113 (Official Build) (64-bit)

**Comment by pawelfus** [comment]
Hi @bhaskar20 

Navigator series is linked to the main series including visibility state. Let's take a look at this example: http://jsfiddle.net/4qv278zf/1/ - as you can see, hiding series from legend hides also  the connected navigator series. This is expected behaviour.

I see you have different jsFiddle and code snippet above. Are you sure you posted correct demo?

**Comment by bhaskar20** [comment]
hi @pawelfus 
I am sorry, I think i forgot to save that jsfiddle, here is another fiddle http://jsfiddle.net/toufu__/yzd148wt/

I understand that navigator series is linked to the first series which does not have showInNavigator option set explicitly as false.
in the above fiddle i have explicitly set showInNavigator as false and am passing series object in navigator option for the chart
```
series: [{
  data: data,
  dataGrouping: {
   enabled: false
   },
   showInNavigator: false
}]
```

```
navigator: {
  adaptToUpdatedData: false,
  series: {
    data: MSFT,
    visible: false
    }
}
```

Here visible option in navigator series option is not honored, and same is not clear from the documentation.
Also, if i donot set showInNavigator in navigator series, series is linked to first series, with visible option from navigator series set on the first render but when you click legend to hide first series and again click on legend to show first series, series is navigator is visible which is inconsistent.
http://jsfiddle.net/toufu__/ob824utj/

**Comment by pawelfus** [comment]
Thank you for the updated demo. 

Regarding issue with separate series for the navigator - it may be a bug. `visible` option should be  applied. I see in the source code that `visible` is forced in the options for the standalone series. I think this is connected to the extremes in the navigator. Without series in the navigator, we can not calculate extremes and the navigator won't work. For example: http://jsfiddle.net/yzd148wt/3/ - programatically hidden series, navigator has no extremes so we can not use it.

However, if you don't want to show series in the navigator, then I would suggest to remove series (leave empty data) and use `navigator.xAxis.min` and `navigator.xAxis.max`. Take a look: http://jsfiddle.net/yzd148wt/2/ 

Regarding visibility for linked series, as I said above, series in navigator is controlled by main series visibility - if that main series' visibility is changed, the same applies for the series in the navigator. I think that `visible` should not be available in the API for the navigator series.

**Comment by pawelfus** [comment]
Issue was fixed in v6.0.0: http://jsfiddle.net/BlackLabel/yzd148wt/32/

Navigator series visibility is not affected by main series, if it's separated. Demo: http://jsfiddle.net/BlackLabel/yzd148wt/34/
